File: Easy/Python/2164. Sort Even and Odd Indices Independently.py
class Solution(object):
    def sortEvenOdd(self, nums):
        # Evens
        for i in range(0, len(nums), 2):
            for j in range(i + 2, len(nums), 2):
                if nums[j] <= nums[i]:
                    temp = nums[i]
                    nums[i] = nums[j]
                    nums[j] = temp
        
        # Odds
        for i in range(1, len(nums), 2):
            for j in range(i + 2, len(nums), 2):
                if nums[i] <= nums[j]:
                    temp = nums[i]
                    nums[i] = nums[j]
                    nums[j] = temp

        return nums

File: Easy/Java/2164. Sort Even and Odd Indices Independently.java
class Solution {
    public int[] sortEvenOdd(int[] nums) {
        // Evens
        for(int i = 0; i < nums.length; i += 2){
            for(int j = i + 2; j < nums.length; j += 2){
                if(nums[j] <= nums[i]){
                    int temp = nums[i];
                    nums[i] = nums[j];
                    nums[j] = temp;
                }
            }
        }

        // Odds
        for(int i = 1; i < nums.length; i += 2){
            for(int j = i + 2; j < nums.length; j += 2){
                if(nums[i] <= nums[j]){
                    int temp = nums[i];
                    nums[i] = nums[j];
                    nums[j] = temp;
                }
            }
        }

        return nums;
    }
}
